src/Parse/ParsePolygon.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Parse;
+
+use Parse\Internal\Encodable;
+use Parse\ParseGeoPoint;
+
+/**
+ * ParsePolygon - Representation of a Parse Polygon object.
+ *
+ * @author Diamond Lewis <findlewis@gmail.com>
+ */
+class ParsePolygon implements Encodable
+{
+    /**
+     * The coordinates.
+     *
+     * @var array
+     */
+    private $coordinates;
+
+    /**
+     * Create a Parse Polygon object.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     */
+    public function __construct($coords)
+    {
+        $this->setCoordinates($coords);
+    }
+
+    /**
+     * Set the Coordinates value for this Polygon.
+     *
+     * @param array $coords GeoPoints or Coordinates.
+     *
+     * @throws ParseException
+     */
+    public function setCoordinates($coords)
+    {
+        if (count($coords) < 3 || !is_array($coords)) {
+            throw new ParseException('Polygon must have at least 3 points');
+        }
+        $points = [];
+        foreach ($coords as $coord) {
+            $geoPoint = $coord;
+            if (is_array($coord)) {
+                $geoPoint = new ParseGeoPoint($coord[0], $coord[1]);
+            }
+            $points[] = [$geoPoint->getLatitude(), $geoPoint->getLongitude()];
```

Comment:
A note that currently this would fail if you pass something that's not an array by accident, like a string, as it would try to run `$geoPoint->get...` on an object that's not necessarily a GeoPoint. If you add the check mentioned just above this shouldn't be a problem anymore though, as you would be able to safely create a new **ParseGeoPoint** and add it after the check.

It looks like you're expecting users to either supply an array of coords or a `ParseGeoPoint` instance. To that end you could add an additional check as an `else if` to ensure that if the object is not an array it is at least an instance of `ParseGeoPoint`. If `$coords` is neither of those it would be safe to throw an exception stating that one or the other is required to proceed.